pkgs/by-name/gh/ghdl/test-simple.nix

```diff
@@ -1,19 +1,22 @@
-{ stdenv, ghdl-llvm, ghdl-mcode, backend }:
+{ stdenv, lib, ghdl-llvm, ghdl-mcode, ghdl-gcc, backend }:
 
 let
-  ghdl = if backend == "llvm" then ghdl-llvm else ghdl-mcode;
+  ghdl = if backend == "llvm" then ghdl-llvm else (if backend == "gcc" then ghdl-gcc else ghdl-mcode);
 in
 stdenv.mkDerivation {
   name = "ghdl-test-simple";
   meta.timeout = 300;
   nativeBuildInputs = [ ghdl ];
+
+  LIBRARY_PATH = lib.optionalString (backend == "gcc") "${stdenv.cc.libc}/lib";
+
```

Comment:
It's not great that setting `LIBRARY_PATH` is necessary for `ghdl-gcc` to be used, since users won't know about that. This is what the gcc wrapper normally fixes. In this case, it would be good to wrap the ghdl executable in `ghdl-gcc` to set `LIBRARY_PATH`. See https://gist.github.com/CMCDragonkai/9b65cbb1989913555c203f4fa9c23374 for an example. That would ensure that `nix-shell -p ghdl-gcc` would just work.
```suggestion
```